sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1406,8 +1406,16 @@ public boolean apply(@Nonnull String input) {
                       klass, entry.getKey(), closestMatches));
           }
         }
-
+        // We need to pull the method directly from klass, to retain generic information.
         Method method = propertyNamesToGetters.get(entry.getKey());
+        String methodName = method.getName();
+        try {
+          method = klass.getMethod(methodName);
+        } catch (NoSuchMethodException e) {
+          // If the method is not found, fall back to just using the registered value.  This
```

Comment:
1. update `getPropertyDescriptors(Class<?> beanClass)` to instead use `allInterfaceMethods` instead of the beanclass. This will help you further along when the proxyinvocationhandler ensures pipeline options is serializable/deserializable and you need the typing information again.

2. If the logic is updated to have the appropriate read/write method stored in the property descriptors then this will fallback to the existing behavior that unknown "properties" fail parsing unless strict parsing is disabled and then we just print out warnings.